### PR TITLE
ExperimentView: select all and sortable

### PR DESCRIPTION
**Pull request by ToonKBC**
Added a checkbox in the experiment view for selecting all runs

### CODE REVIEW

**Comment by ToonKBC** [comment]
I'll also add the ability to sort by a metric or parameter. Issue #143.

**Comment by mateiz** [comment]
This is awesome, but I noticed a few small issues:

1) Can you make the default order be decreasing by date (i.e. show the newest run at the top)? I think you now show the oldest run at the top, which is a change from the current UI.

2) Can you add `white-space: nowrap` to the column headers? It will prevent this ugly behavior where the caret leaks onto a separate line:
screen shot 2018-08-01 at 2 34 27 pm

3) I would also add `cursor: pointer` to the column headers to make it clearer that they're clickable. 

4) I saw that some columns, such as User and Source, don't support sorting. Why not enable it on all of them?

File: mlflow/server/js/src/components/ExperimentView.js [hunk 1]
```diff
@@ -87,20 +93,50 @@ class ExperimentView extends Component {
...
+      }
+    });
+    rows.sort((a, b) => {
+      let x = a.sortValue;
+      let y = b.sortValue;
+      if(x === undefined)
+        return 1;
+      if(y === undefined)
+        return -1;
+      if(!this.state.sort.ascending) {
```
Comment (mateiz) [inline mlflow/server/js/src/components/ExperimentView.js]: Nit: try to follow the coding style for the other code here, which is to leave a space after `if`

**Comment by mateiz** [comment]
Thanks for the fixes. For some reason, sorting by "version" doesn't work though, which is weird (nothing changes). I also noticed a few React warnings in the console, such as:

> Warning: Failed prop type: ExperimentView: prop type `searchInput` is invalid; it must be a function, usually from the `prop-types` package, but received `undefined`.

>Warning: React does not recognize the `preventDefault` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `preventdefault` instead. If you accidentally passed it from a parent component, remove it from the DOM element."

But I'm not sure whether these are all caused by this patch (they might have been there before).

**Comment by mateiz** [comment]
BTW if you'd like, I can try to fix those issues and merge it. Overall the functionality is great.

**Comment by ToonKBC** [comment]
The render method in utils was used but as sortValue. This gave some weird sorting quirks with dom-object, I think.

The warnings were there already before. A whole new pull request will be needed to fix all react warnings (and errors).

**Pull request merged**